.github/workflows/upload-release-to-plugin-repo.yml

```
@@ -159,3 +174,57 @@ jobs:
                   sed -i "s/Stable tag: ${STABLE_VERSION_REGEX}/Stable tag: ${VERSION}/g" ./readme.txt
                   svn commit -m "Releasing version $VERSION" \
                    --no-auth-cache --non-interactive  --username "$SVN_USERNAME" --password "$SVN_PASSWORD"
+
+    upload-branch:
+        name: Publish as branch (and tag)
+        runs-on: ubuntu-latest
+        environment: wp.org plugin
+        needs: [compute-latest-stable, update-changelog]
+        if: |
+            !endsWith( github.ref, needs.compute-latest-stable.outputs.latest_stable_tag ) &&
+            !github.event.release.prerelease && github.event.release.assets[0]
+        env:
+            PLUGIN_REPO_URL: 'https://plugins.svn.wordpress.org/gutenberg'
+            STABLE_VERSION_REGEX: '[0-9]\+\.[0-9]\+\.[0-9]\+\s*'
+            SVN_USERNAME: ${{ secrets.svn_username }}
+            SVN_PASSWORD: ${{ secrets.svn_password }}
+            VERSION: ${{ github.event.release.name }}
+
+        steps:
+            - name: Check out Gutenberg branches from WP.org plugin repo
+              run: svn checkout "$PLUGIN_REPO_URL/branches" --username "$SVN_USERNAME" --password "$SVN_PASSWORD"
+
+            - name: Download and unzip Gutenberg plugin asset into trunk folder
+              env:
+                  PLUGIN_URL: ${{ github.event.release.assets[0].browser_download_url }}
+              run: |
+                  # do the magic here
+                  curl -L -o gutenberg.zip $PLUGIN_URL
+                  unzip gutenberg.zip -d branches/$VERSION
+                  rm gutenberg.zip
+
+            - name: Replace the stable tag placeholder with the existing stable tag on the SVN repository
+              env:
+                  STABLE_TAG_PLACEHOLDER: 'Stable tag: V\.V\.V'
+              run: |
+                  sed -i "s/$STABLE_TAG_PLACEHOLDER/Stable tag: $VERSION/g" ./branches/$VERSION/readme.txt
+
+            - name: Download Changelog Artifact
+              uses: actions/download-artifact@9782bd6a9848b53b110e712e20e42d89988822b7 # v3.0.1
+              with:
+                  name: changelog trunk
+                  path: branches/$VERSION
```

Not sure this is working: I have a `branches/$VERSION/changelog.txt` in my svn now. (I.e. with a folder that's literally named `$VERSION`.)

I think variable substitution works differently in GH workflows/YAML files -- normal shell substitution only works in `run:` ('cause that's normally executed in a shell 😅 ). Everywhere else, we need `${{ }}` for [expressions](https://docs.github.com/en/actions/learn-github-actions/expressions), and env [variables](https://docs.github.com/en/actions/learn-github-actions/variables) aren't directly available (but through `env.xyz`, if I'm not mistaken). The following should work though:

```suggestion
                  path: branches/${{ github.event.release.name }}
```